packages/actions-spec/index.d.ts

```diff
@@ -73,27 +73,91 @@ export interface LinkedAction {
   /** button text rendered to the user */
   label: string;
   /** parameters used to accept user input within an action */
-  parameters?: ActionParameter[];
+  parameters?: TypedActionParameter[];
 }
 
+export type TypedActionParameter<
+  T extends ActionParameterType = ActionParameterType,
+> = T extends SelectableParameterType
+  ? ActionParameterSelectable<T>
+  : ActionParameter<T>;
+
 /**
  * Parameter to accept user input within an action
  */
-export interface ActionParameter {
+export interface ActionParameter<T extends ActionParameterType, M = MinMax<T>> {
+  /** input field type */
+  type?: T;
   /** parameter name in url */
   name: string;
   /** placeholder text for the user input field */
   label?: string;
   /** declare if this field is required (defaults to `false`) */
   required?: boolean;
+  /** regular expression pattern to validate user input client side */
+  pattern?: string;
+  /** human-readable description of the `type` and/or `pattern`, represents a caption and error, if value doesn't match */
+  patternDescription?: string;
+  /** the minimum value allowed based on the `type` */
+  min?: M;
+  /** the maximum value allowed based on the `type` */
+  max?: M;
+}
+
+type MinMax<T extends ActionParameterType> = T extends "date" | "datetime-local"
+  ? string
+  : T extends "radio" | "select"
+  ? never
+  : number;
+
+type GeneralParameterType =
```

Comment:
Let's make these exportable as well, so that it would be easy to narrow types (without redefining them)

Reply:
Can do!